File: desktop/scripts/check_browser_errors.js
import { chromium } from "playwright";
import fs from "fs";

async function main() {
  const browser = await chromium.launch();
  const page = await browser.newPage();
  
  page.on("console", msg => {
    if (msg.type() === "error") {
      console.log(`[Browser Error]: ${msg.text()}`);
    } else if (msg.type() === "warning") {
      console.log(`[Browser Warning]: ${msg.text()}`);
    } else {
      console.log(`[Browser Console]: ${msg.text()}`);
    }
  });

  page.on("pageerror", exception => {
    console.log(`[Uncaught Exception]: ${exception}`);
  });

  // Navigate to local dev server
  await page.goto("http://localhost:1420");
  
  // Wait to see if error occurs
  await page.waitForTimeout(3000);
  
  await browser.close();
}

main().catch(console.error);

File: check_browser_errors.py
from playwright.sync_api import sync_playwright
import time

def run():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        
        def handle_console(msg):
            print(f"[Console] {msg.type}: {msg.text}")
            
        def handle_pageerror(err):
            print(f"[PageError] {err}")

        page.on("console", handle_console)
        page.on("pageerror", handle_pageerror)
        
        print("Navigating to localhost:1420...")
        try:
            page.goto("http://localhost:1420")
            time.sleep(5)
            # Take a screenshot
            screenshot_path = "system/logs/blank_screen_test.png"
            page.screenshot(path=screenshot_path)
            print(f"Screenshot saved to {screenshot_path}")
        except Exception as e:
            print(f"Error: {e}")
        
        browser.close()

if __name__ == "__main__":
    run()
